contracts/proxies/CurrencyConvertor.sol

```diff
@@ -115,6 +121,7 @@ contract CurrencyConvertor is BaseRelayRecipient {
     * @param  positionId         The positionId of the L2 account to deposit into.
     * @param  exchange           The exchange being used to swap the taker token for USDC.
     * @param  allowanceTarget    The address being approved for the swap.
+    * @param  signature          The signature for registering. NOTE: if length is 0, will not try to register
```

Comment:
Any reason signature is here? I think you should swap `signature` and `data`. It's better for optional variables (`signature`) to come last. Also `data` is logically related to the other params so it's weird for it to be separated.